src/pax_store/Page.h

```diff
@@ -0,0 +1,248 @@
+        // Array of all row indices which have passed all filters that were already applied. At the beginning, all rows all still present
+        // because no filter has been applied yet. However, for performance reasons, we do not populate the array with all indicies but
+        // rather compute the indices on the fly when the first filter is applied (firstRun=true).
+        // Please note that, for performance reasons, we do not recreate the positions array after each iteration. Instead, we overwrite the
+        // elements starting from the beginning of each iteration and keep track of all still valid positions using the `positionsCounter`.
+        // All values with an index greater than `positionsCounter` are outdated intermediate results.
+        auto positions = new uint64_t[*numberOfRecords];
+        bool firstRun = true;
+
+        // Count of all rows which have passed all filters.
+        uint64_t positionsCounter = *numberOfRecords;
+
+        // Batch size. Equals the number of cells that fit in one AVX512 register (512bites = 64bytes).
+        auto rowsPerRegister = (unsigned)64 / sizeof(T);
+
+        for (auto &filter : filters) {
+            // Find the minipage for the attribute to which the filter applies.
+            Header offset = *(minipageOffsets + filter->index);
+            T *minipage = (T *)((char *)start + offset);
+
+            auto *positionsTail = positions;
+
+            // Counter of all rows which have also passed the current filter.
+            uint64_t newPositionsCounter = 0;
+
+            for (RowIndex rowIndex = 0; rowIndex < positionsCounter; rowIndex += rowsPerRegister) {
+                // Check bounds for next load into AVX register.
+                __mmask16 mask;
+                if (rowIndex + rowsPerRegister <= positionsCounter) {
+                    // We are inside the bounds. There is enough data still in the row to fill a whole AVX512 register.
+                    mask = ONE_MASK;
+                } else {
+                    // We are close to the end of the row and there is not enough data left to fill a whole AVX512 register. Still, we will
+                    // load as much data as needed to fill the whole register, but we tell the filter to ignore the random data that pads
+                    // the register.
+                    auto remaining = positionsCounter % rowsPerRegister;
+                    mask = _mm512_int2mask(~(0xffffffff << remaining));
+                }
+
+                // Load the indices of the rows
+                uint64_t *vindex;
+                std::pair<__m512i, __m512i> loadResult;
+                if (firstRun) {
+                    // On the first run, the positions array contains only zeros. Since all rows are still valid, we can simply load the
+                    // next batch of rows starting from the current `rowIndex`.
+                    loadResult = ColumnStore::Helper::load(minipage, rowIndex);
```

Comment:
Nitpick: `ColumnStore::Helper::load` and `ColumnStore::Helper::gather` have the arguments in the opposite order. Do you mind if swap them for one the functions?

Reply:
Yes you can swap them, this is an error from early development of these methods. I propose that we use the parameter list of `ColumnStore::Helper::gather` (first the index, then the data).

Reply:
Will keep that in mind for later, but merge now to have `master` working again